packages/@o3r/rules-engine/schematics/rules-engine-to-component/index.ts

```diff
@@ -0,0 +1,138 @@
+import { chain, noop, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import {
+  applyEsLintFix,
+  findMethodByName,
+  fixStringLiterals,
+  generateBlockStatementsFromString,
+  generateClassElementsFromString,
+  generateImplementsExpressionWithTypeArguments,
+  getLibraryNameFromPath,
+  getSimpleUpdatedMethod, sortClassElement
+} from '@o3r/schematics';
+import { insertImport } from '@schematics/angular/utility/ast-utils';
+import { applyToUpdateRecorder, Change } from '@schematics/angular/utility/change';
+import * as ts from 'typescript';
+import { NgGenerateRulesEngineToComponentSchematicsSchema } from './schema';
+
+/**
+ * Generate the code to enable rules-engine on a component
+ *
+ * @param options
+ */
+export function ngGenerateRulesEngineToComponent(options: NgGenerateRulesEngineToComponentSchematicsSchema): Rule {
```

Comment:
You should add a check to throw if the component already have the rules engine

Reply:
done